objects/client.c

```diff
@@ -969,6 +969,8 @@ client_focus_update(client_t *c)
         luaA_object_emit_signal(L, -1, "focus", 0);
 
     lua_pop(L, 1);
+    
+    ewmh_update_net_current_desktop();
```

Comment:
Instead of having callbacks everywhere (e.g. here), could you connect to signals? In function `ewmh_init` various callbacks are connected and this even already connects a callback to the "focus" signal. You can just copy that line and connect `ewmh_update_net_current_desktop`. (same for other places where you add callbacks: Connect to `property::screen` on clients,